clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClansMovementListener.java

```diff
@@ -109,27 +111,45 @@ public void displayOwner(Player player, Clan locationClan) {
     public void onClanHomeTeleport(ClanHomeTeleportEvent event) {
         if (event.isCancelled()) return;
 
+        Player player = event.getPlayer();
+
+        if (!clanManager.canTeleport(player)) {
+            UtilMessage.message(player, "Clans", "You cannot teleport while combat tagged.");
+            event.setCancelled(true);
+            return;
+        }
+
         clanManager.getClanByLocation(event.getPlayer().getLocation()).ifPresentOrElse(clan -> {
             if (clan.isAdmin()) {
                 if (clan.isSafe() && clan.getName().contains("Spawn") && event.getPlayer().getLocation().getY() > 110) {
                     return;
                 }
             }
 
-            UtilMessage.message(event.getPlayer(), "Clans", "You can only teleport to your clan home from spawn or the wilderness.");
-            event.setCancelled(true);
+            if (clanManager.getRelation(clanManager.getClanByPlayer(event.getPlayer()).orElse(null), clan) == ClanRelation.ENEMY) {
+                UtilMessage.message(event.getPlayer(), "Clans", "You cannot teleport to your clan home from enemy territory.");
+                event.setCancelled(true);
```

Comment:
Maybe make this the same duration as c stuck (2 minutes)

Reply:
Should I then make it 1 minute in any other territory aswell? And if we are allowing it In shops now it should probably be instant there given that you arent combat tagged.

Reply:
did this